### PR TITLE
Fix some derived type issues

### PR DESCRIPTION
**Pull request by Thirumalai-Shaktivel**
db8bfc5c7a85d36d3c81afa6bbcf925f70516945 and 427c7c2bc9dd0047884f582e045bc5c073f8737e

### CODE REVIEW

File: src/lfortran/pickle.cpp [hunk 1]
```diff
@@ -156,6 +156,22 @@ class PickleVisitor : public PickleBaseVisitor<PickleVisitor>
...
+        s.append("[");
+        for (size_t i=0; i<x.n_member; i++) {
+            this->visit_struct_member(x.m_member[i]);
+            if (i < x.n_member-1) s.append(" ");
+        }
+        s.append("]");
+        if(indent) {
+            dec_indent();
+            s.append("\n" + indtd);
+        }
```
Comment (certik) [inline src/lfortran/pickle.cpp:174]: I would create a separate PR just for this change, to keep the number of changes in tests minimal in this PR.

**Comment by certik** [comment]
Otherwise this looks good.

File: src/lfortran/semantics/ast_common_visitor.h [hunk 2]
```diff
@@ -1343,7 +1343,23 @@ class CommonVisitor : public AST::BaseVisitor<Derived> {
...
+                            func_call->m_func);
+                        if (sym_found == nullptr) {
+                            std::string type_name(func_call->m_func);
+                            throw SemanticError("Derived type `" + type_name +
+                                "` is not defined", func_call->base.base.loc);
+                        }
+                    } else {
+                        throw SemanticError("Only function call assignment is allowed for now",
+                            x.base.base.loc);
+                    }
```
Comment (Thirumalai-Shaktivel) [inline src/lfortran/semantics/ast_common_visitor.h:1361]: Here we do not store (in ASR) any information about the initializer.
Instead, we make some semantics checks.

**Comment by Thirumalai-Shaktivel** [comment]
This PR is ready to go in.

File: src/lfortran/pickle.cpp (hunk 1 above)
Comment (Thirumalai-Shaktivel) [reply src/lfortran/pickle.cpp]: https://github.com/lfortran/lfortran/pull/1121

**Review by certik — APPROVED** [review]
I think this is good. Conflicts must be resolved.

**Pull request merged**